enhancements/baremetal-config.md

```diff
@@ -0,0 +1,126 @@
+to the Infrastructure Config CR. It is also aware of the additional config
+required during the deployment of a metal3 cluster. This enhancement proposes
+to extend the BareMetaPlatformStatus CR to hold these additional values needed
+for a metal3 deployment.
+
+## Motivation
+
+Adding these additional config items to the BareMetalPlatformStatus CR, allows
+versioning of these config items.
+
+### Goals
+
+The goal of this enhancement is to enable the installer to add vital config
+values for a Baremetal IPI deployment to a CR. The MAO will then be able to
+access these values when it kicks off a metal3 cluster.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
+
+### Proposal
+
+To support configuration of a Baremetal IPI (metal3) deployment, a set of new
+config items are required to be made availabe via a Config Resource. The
+BareMetalPlatformStatus is an ideal fit because it contains configuration
+that is specific to a Baremetal deployment.
+
+This proposal aims to add the following to BareMetalPlatformStatus:
+
+1. ProvisioningInterface : This is the interface name on the underlying
+baremetal server which would be physically connected to the provisioning
+network.
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on
+the baremetal server for provisioning. This value should not be in the
+DHCP range and should not be used in the provisioning network for any
+other purpose (should be a free IP address.) It is expected to be provided
+in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its
+CIDR. The ProvisioningIP and the ProvisioningDHCPRange are expected to
+be within this network. The value for this config is expected to be
+provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which
+baremetal hosts can be assigned their IP address. This range should be
+provided as two comma seperated IP addresses where the addresses represent
+a contiguous range of IPs that are available to be allocated. It is expected
+to be provided in the format : 10.1.0.10, 10.1.0.100.
+
+5. CacheURL : This is a location where image required to bring up baremetal
+hosts have been previously downloaded for faster image downloads within the
+cluster. This is an optional parameter and is expected to be provided
+in the format : http://111.111.111.1/images
+
+6. RHCOSImageURL : This config is expected to be the location of the
+official RHCOS release image. It is a required parameter and will be used
+to deploy new nodes in the cluster.
+
+### User Stories [optional]
+
+This enhancement will allow users to have their Barametal IPI deployments
+customizable to their specific hardware and network requirements.
```

Comment:
It also allows IPI baremetal deployments without manual workarounds such as creating the required ConfigMap object manually (which is what we're currently doing for testing).

Reply:
The user story doesn't actually provide the components that will be using these fields.

Since this proposal is adding these new fields to the _global_ platform configs, I would like to make sure this is actually a infrastructure level information or it belongs to a specific operator.

Reply:
Augmented The User Stories section to take care of the above 2 comments.